### DISCUSSION TITLE
Does it support ios with IL2CPP

### DISCUSSION THREAD

**Discussion by hello32768** [exodrifter/unity-python#7] [Q&A] [answered]
Hi, does it support ios with IL2CPP, Unity 2017 doesn't support mono on ios now.

**Comment by exodrifter** [comment]
As far as I know, the compatibility of DLLs shouldn't be effected by what scripting backend you use. This plugin should work on iOS.

**Comment by sekkit** [comment]
Tested old version on iOS, il2cpp won't compile, with few modifications it compiled.
BUT will throw runtime error on iOS due to method not found exception.
il2cpp won't allow create new types because it's pure c++.
https://docs.unity3d.com/Manual/IL2CPP-BytecodeStripping.html
IronPython is based on DLR which is dynamic typing.
Type Reflection must be declared at least once or il2cpp won't recoginized it.
DLR needs some modification to run on iOS, maybe you can do that.

**Comment by exodrifter** [comment]
I fixed the compilation error you mentioned, but I wasn't able to reproduce your runtime error in the editor with the iOS platform selected. Does your runtime error happen on an iOS device? Can you give me a more detailed error?

I'm also not sure what you're trying to say about dynamic typing not working with il2cpp because it's pure c++ and then saying that the solution lies in Bytecode Stripping. Could you please explain?

As to answering the original question, you can use the latest version of the library and make a build for iOS with the IL2CPP scripting backend with no problem. I don't have an iOS device though, so I can't test if it actually runs.

**Comment by sekkit** [comment]
Inside editor, only mono runtime is used.
il2cpp means all types, function decleration, must be declared explicitly. If you create a new type at runtime, it's forbidden in iOS. 
I think when compile the IronPython code, dlr generated file xxx.generated.cs can be modified to fit il2cpp requirement.
You can run it in a xcode iOS simulator, no real device required. (Unity build settings->target SDK, Simulator SDK)
I can provide u with no detailed errors because that's months ago, I did a thorough test of that. You can test it directly on simulator.

**Comment by exodrifter** [comment]
Unfortunately, I don't have a MacBook, so I can't run the simulator either. I won't be able to resolve this issue for the foreseeable future.

**Comment by sekkit** [comment]
you don't need a mac to test, for il2cpp now supports many other platforms.

**Comment by joaocc** [comment]
An interesting test could be to check if this runs on 2019.x IL2CPP for Android or WSA/Hololens... These are prob as bit less constrained than iOS and easier to test on Linux/Windows PCs

**Comment by exodrifter** [comment]
Perhaps. But, that sounds like a different issue and knowing the answer to that issue doesn't necessarily answer the question "does this work in iOS when you use the IL2CPP backend".

**Comment by joaocc** [comment]
@exodrifter you are correct. But I was thinking it would answer to another point that was raised in the thread about the impossibility of running on IL2CPP (sorry for forgetting to make it explicit). 
We are looking at this from the perspective of assessing viability for Hololens, which also means IL2CPP. Thx

**Comment by exodrifter** [comment, marked as answer]
After doing more research and remembering that this issue is still open, there are two things that are very clear to me now:

* IronPython _requires_ `System.Reflection.Emit` to work and does not have support for IL2CPP compilation anyway (see IronLanguages/ironpython2#258)
* Unity does not support `System.Reflection.Emit` on IL2CPP backends ([source](https://docs.unity3d.com/2019.4/Documentation/Manual/ScriptingRestrictions.html))

So the answer is: No, it doesn't support iOS with the IL2CPP backend. I have updated the README accordingly in bb595d4e2c763535e058243df9e9e1d14fc2188c.
